kairon/evaluator/main.py

```diff
@@ -0,0 +1,325 @@
+    ValidationError,
+    OperationError,
+    NotRegistered,
+    InvalidDocumentError,
+    LookUpError,
+    MultipleObjectsReturned,
+    InvalidQueryError,
+)
+from pymongo.errors import PyMongoError
+from secure import StrictTransportSecurity, ReferrerPolicy, ContentSecurityPolicy, XContentTypeOptions, Server, \
+    CacheControl, Secure, PermissionsPolicy
+from starlette.exceptions import HTTPException as StarletteHTTPException
+
+from kairon.shared.utils import Utility
+from kairon.api.models import Response
+from kairon.exceptions import AppException
+from kairon.evaluator.router import pyscript
+
+logging.basicConfig(level="DEBUG")
+hsts = StrictTransportSecurity().include_subdomains().preload().max_age(31536000)
+referrer = ReferrerPolicy().no_referrer()
+csp = (
+    ContentSecurityPolicy().default_src("'self'")
+        .frame_ancestors("'self'")
+        .form_action("'self'")
+        .base_uri("'self'")
+        .connect_src("'self'" "api.spam.com")
+        .frame_src("'self'")
+        .img_src("'self'", "static.spam.com")
+)
+cache_value = CacheControl().must_revalidate()
+content = XContentTypeOptions()
+server = Server().set("Secure")
+permissions_value = (
+    PermissionsPolicy().accelerometer("").autoplay("").camera("").document_domain("").encrypted_media("")
+        .fullscreen("").geolocation("").gyroscope("").magnetometer("").microphone("").midi("").payment("")
+        .picture_in_picture("").sync_xhr("").usb("").geolocation("self", "'spam.com'").vibrate()
+)
+secure_headers = Secure(
+    server=server,
+    csp=csp,
+    hsts=hsts,
+    referrer=referrer,
+    permissions=permissions_value,
+    cache=cache_value,
+    content=content
+)
+
+app = FastAPI()
+app.add_middleware(
+    CORSMiddleware,
+    allow_origins=["*"],
+    allow_credentials=True,
+    allow_methods=["*"],
+    allow_headers=["*"],
+    expose_headers=["content-disposition"],
+)
+app.add_middleware(GZipMiddleware)
+Utility.load_environment()
+apm_client = Utility.initiate_fastapi_apm_client()
+if apm_client:
+    app.add_middleware(ElasticAPM, client=apm_client)
+
+
+@app.middleware("http")
+async def add_secure_headers(request: Request, call_next):
+    """Add security headers."""
+    response = await call_next(request)
+    secure_headers.framework.fastapi(response)
+    return response
+
+
+@app.middleware("http")
+async def log_requests(request: Request, call_next):
+    """Logging request calls."""
+    authorization: str = request.headers.get("Authorization")
+    _, param = get_authorization_scheme_param(authorization)
+    start_time = time()
+
+    response = await call_next(request)
+
+    process_time = (time() - start_time) * 1000
+    formatted_process_time = "{0:.2f}".format(process_time)
+    logger.info(
+        f"rid={param} request path={request.url.path} completed_in={formatted_process_time}ms status_code={response.status_code}"
+    )
+
+    return response
+
+
+@app.exception_handler(StarletteHTTPException)
```

Comment:
instead of using multiple exception handlers, use single one that handles all: https://github.com/digiteinfotech/kairon/blob/master/kairon/events/server.py#L87